### PR TITLE
Various Bug fixes

### LINKED ISSUE #216: There isnt a way to easily change my primary photo or re-order my photos on a Listing
The photos currently display in the order they were uploaded. It would be great if you could drag and drop photos to change the order. We're also going to need to display an indicator on the primary photo.

screenshot 2016-12-15 12 58 09

### LINKED ISSUE #259: Background click event should not close modals
We discussed removing this behavior on #257

### LINKED ISSUE #261: Edit listing modal does not auto scroll to first error.
If attempting to save a listing and a client side error occurs, the modal does not automatically scroll to the first error.

This was in place before, but somehow got broken, likely in the modal style refactor.

### LINKED ISSUE #266: Remove the Shipping placeholder in Create Listing
@rmisio @jjeffryes I commented this section out (pictured below). We no longer need it since we have the **Shipping Option 1** placeholder. Is removing it going to introduce any issues? Can we completely remove it from the code? 
screenshot 2017-01-04 13 56 03

This still exists
screenshot 2017-01-04 13 53 34

### LINKED ISSUE #267: Username in SMTP notification settings must allow email address
Usernames for SMTP servers can be email addresses and currently we only allow alphanumeric.

image

### PR DESCRIPTION
**Pull request by rmisio**
closes #261 
closes #267 
closes #216 
closes #259 
closes #266

### CODE REVIEW

**Comment by morebrownies** [comment]
Hey @rmisio when hovering a photo in Edit Listing could we change the cursor to cursor:move? That should make it more clear that they can be dragged.

**Comment by rmisio** [comment]
@morebrownies sure, done.

**Comment by morebrownies** [comment]
👍

File: js/templates/modals/editListing/uploadPhoto.html [hunk 1]
```diff
@@ -1,4 +1,7 @@
 <li style="background-image: url('<%= ob.getServerUrl(`ipfs/${ob.small}`) %>')" class="tile">
+  <div class="primaryIndicatorWrap">
+    <div class="clrE1 clrTOnEmph phraseBox padTn">PRIMARY PHOTO</div>
```
Comment (jjeffryes) [inline js/templates/modals/editListing/uploadPhoto.html]: "PRIMARY PHOTO" should have a translation.

File: js/templates/modals/editListing/editListing.html [hunk 2]
```diff
@@ -150,16 +150,6 @@ <h2 class="h4 clrT"><%= ob.polyT('editListing.sectionNames.photos') %></h2>
       </section>
 
       <section class="js-scrollToSection js-sectionShipping shippingSection <% if (ob.listing.metadata.contractType !== 'PHYSICAL_GOOD') print('disabled') %> <% if (ob.listing.shippingOptions.length) print('hasShippingOptions') %>">
```
Comment (jjeffryes) [inline js/templates/modals/editListing/editListing.html]: Don't forget to remove hasShippingOptions here.

File: js/views/modals/editListing/EditListing.js [hunk 3]
```diff
@@ -875,18 +837,42 @@ export default class extends BaseModal {
...
+        onUpdate: (e) => {
+          const imageModels = this.innerListing
+            .get('item')
+            .get('images')
+            .models;
+
+          imageModels[e.oldIndex - 1] =
+            imageModels.splice(e.newIndex - 1, 1, imageModels[e.oldIndex - 1])[0];
+        },
+      });
```
Comment (jjeffryes) [inline js/views/modals/editListing/EditListing.js]: The sortable code lets you move the add photo button, it seems like that should be excluded. 

image

File: styles/modules/modals/_editListing.scss [hunk 4]
```diff
@@ -39,6 +39,10 @@
 
     .tabContent {
       .photoUploadSection {
+        h2 {
+          display: inline-block;
```
Comment (jjeffryes) [inline styles/modules/modals/_editListing.scss]: What does this do? It doesn't seem to change anything in the layout

Comment (rmisio) [inline styles/modules/modals/_editListing.scss]: During an upload, if that h2 isn't an inline-block, then the "uploading" status gets a scrollbar.

image

I know I could probably re-work this to use flex, but I'll be gosh darned if I'm going to use flex for ever little think... lol!

Comment (jjeffryes) [inline styles/modules/modals/_editListing.scss]: You could just put a floL class on the h2, and then you wouldn't need this rule.

**Comment by rmisio** [comment]
ok @jjeffryes I've fixed the issues you found (good catches).

**Pull request merged**